libmscore/rest.cpp

```diff
@@ -429,7 +429,7 @@ void Rest::checkDots()
             dot->setVisible(visible());
             score()->undoAddElement(dot);
             }
-      if (n < 0) {
+      if (n < 0 && _dots.size()) {
```

Comment:
That wouldn't probably help if `-n > _dots.size()` and `_dots` is not empty. It looks like this check should be moved into the loop instead.
Also some `qDebug()` or `qWarning()` message for this case could perhaps be useful, this situation shouldn't normally happen.

Reply:
I like the other fix better anyhow, more fixing it at the deepest root

Reply:
You mean something like

      if (n < 0) {
            for (int i = 0; i < -n; ++i) {
                  if (!_dots.size() || -n > _dots.size()) {
                        qDebug("no dots at all or number mismatch");
                        return;
                        }
                  score()->undoRemoveElement(_dots.back());
                  }
            }

Reply:
Something like this actually:
```
  if (n < 0) {
        for (int i = 0; i < -n; ++i) {
              if (_dots.empty()) {
                    qWarning("no dots at all or number mismatch");
                    return;
                    }
              score()->undoRemoveElement(_dots.back());
              }
        }
```
This will clear dots if `-n > _dots.size()` which seems sensible to do in this case.

But maybe the fix in #5235 should indeed be enough so let's apply that one first.

Reply:
I don't quite see the difference to my initial change though, except that now we go into the for loop once in the error case, something that the initial fix did not. To me this is just a more expensive way of doing the same thing?

Reply:
There is no difference if there is no dots initially. If there are some dots but `-n` appears to have a large value then the approach I propose will just clear all the dots without crashing the program. But I am not sure whether such situations actually happen.